src/hugr/views.rs

```diff
@@ -300,6 +320,48 @@ pub trait HugrView: sealed::HugrInternals {
     }
 }
 
+/// Wraps an iterator over [Port]s that are known to be [OutgoingPort]s
+#[derive(Debug)]
+pub struct OutgoingPorts<T>(T);
```

Comment:
Alternatively,
```rust
pub type OutgoingPorts<I> = std::iter::Map<I, fn(Port) -> OutgoingPort>
```
and then create it with
```rust
self.node_ports(node, Direction::Outgoing).map(|p| p.as_outgoing().unwrap());
```

Otherwise, you need to derive the other iterator traits: `FusedIterator`, `DoubleEndedIterator`, `ExactSizeIterator`.

Reply:
Nice!! I was hoping to do something like this but somehow got stuck looking at much more complicated `type` definitions than your simple one - moreover, more complicated ones that didn't work!

The other possibility would be to redefine `NodeInputs` as parameterized (over `T: Into<Port>`); and then leave `node_inputs` and `node_outputs` both to subclasses, with `node_ports` picking one or the other and then `.map_into`....no, that doesn't work, because the type of the MapInto depends on the source type, bah. Ok - thanks! :)

Reply:
Also, we could inline the definition of `IncomingNodes` (etc.) - each is only used once now...